staging/src/k8s.io/client-go/tools/remotecommand/websocket.go

```diff
@@ -497,7 +493,7 @@ func (h *heartbeat) start() {
 			// "WriteControl" does not need to be protected by a mutex. According to
 			// gorilla/websockets library docs: "The Close and WriteControl methods can
 			// be called concurrently with all other methods."
-			if err := h.conn.WriteControl(gwebsocket.PingMessage, h.message, time.Now().Add(writeDeadline)); err == nil {
+			if err := h.conn.WriteControl(gwebsocket.PingMessage, h.message, time.Now().Add(2*time.Second)); err == nil {
```

Comment:
Doesn't this write deadline trigger the same connection error state that makes the connection unusable?

> After a write has timed out, the websocket state is corrupt and all future writes will return an error

Reply:
I expected the write deadline to be the max of the time we'd be willing to wait for the connection to recover. It sounds like as soon as we miss a single write deadline, the connection becomes unusable

Reply:
`WriteControl` does not have the same comment about making the connection unusable afterwards. It only says
```
// WriteControl writes a control message with the given deadline. The allowed                                                                                                                          
// message types are CloseMessage, PingMessage and PongMessage.      
```

l'll dig further, but I think this method is not going to ruin the connection if there is a timeout.

Reply:
OK--I think you're right. I should increase this ping write to the `writeDeadline` (if we do not get rid of it entirely by passing an empty time).

Reply:
I think the same timeout as `pingReadDeadline` makes sense... if we can't send a ping for that long, we want to tear down

Reply:
Done